src/webapi/events/drag.rs

```diff
@@ -0,0 +1,697 @@
+    ///
+    /// [(Javascript docs)](https://developer.mozilla.org/en-US/docs/Web/API/DataTransfer/items)
+    pub fn items( &self ) -> DataTransferItemList {
+        js!(
+            return @{self.as_ref()}.items;
+        ).try_into().unwrap()
+    }
+
+    /// Contains a list of all the local files available on the data transfer.
+    /// If the drag operation doesn't involve dragging files, this property is an empty list.
+    ///
+    /// [(Javascript docs)](https://developer.mozilla.org/en-US/docs/Web/API/DataTransfer/files)
+    pub fn files( &self ) -> FileList {
+        js!(
+            return @{self.as_ref()}.files;
+        ).try_into().unwrap()
+    }
+
+    /// An array of strings giving the formats that were set in the dragstart event.
+    ///
+    /// [(Javascript docs)](https://developer.mozilla.org/en-US/docs/Web/API/DataTransfer/types)
+    pub fn types( &self ) -> Vec<String> {
+        js!(
+            return @{self.as_ref()}.types;
+        ).try_into().unwrap()
+    }
+
+    /// Remove the data associated with a given type. The type argument is optional.
+    /// If the type is empty or not specified, the data associated with all types is removed.
+    /// If data for the specified type does not exist, or the data transfer contains no data,
+    /// this method will have no effect.
+    ///
+    /// [(Javascript docs)](https://developer.mozilla.org/en-US/docs/Web/API/DataTransfer/clearData)
+    pub fn clear_data( &self, format: Option<&str> ) {
+        match format {
+            None => js!(@(no_return) @{self.as_ref()}.clearData()),
+            Some(x) => js!(@(no_return) @{self.as_ref()}.clearData(@{x}))
+        };
+    }
+
+    /// Retrieves the data for a given type, or an empty string if data for that type does not exist
+    /// or the data transfer contains no data.
+    ///
+    /// [(Javascript docs)](https://developer.mozilla.org/en-US/docs/Web/API/DataTransfer/getData)
+    pub fn get_data( &self, format: &str ) -> String {
+        js!(
+            return @{self.as_ref()}.getData(@{format});
+        ).try_into().unwrap()
+    }
+
+    /// Set the data for a given type.
+    /// If data for the type does not exist, it is added at the end, such that the last item in the
+    /// types list will be the new format.
+    /// If data for the type already exists, the existing data is replaced in the same position.
+    ///
+    /// [(Javascript docs)](https://developer.mozilla.org/en-US/docs/Web/API/DataTransfer/setData)
+    pub fn set_data( &self, format: &str, data: &str ) {
+        js!(@(no_return)
+            @{self.as_ref()}.setData(@{format}, @{data});
+        );
+    }
+
+    /// Set the image to be used for dragging if a custom one is desired.
+    ///
+    /// [(Javascript docs)](https://developer.mozilla.org/en-US/docs/Web/API/DataTransfer/setDragImage)
+    pub fn set_drag_image( &self, img: &ImageElement, x_offset: u32, y_offset: u32 ) {
```

Comment:
The specs say that the offsets should be `i32`. (Side note: does it actually make sense to pass negative values here?)

Reply:
> The values are expressed as distances in CSS pixels from the left side and from the top side of the image respectively
I don't think it makes sense to pass negative values here, but I'll update to match the spec.